src/main/java/org/reactome/addlinks/fileprocessors/COSMICFileProcessor.java

```diff
@@ -23,17 +25,23 @@ public COSMICFileProcessor(String processorName)
 	@Override
 	public Map<String, String> getIdMappingsFromFile()
 	{
-		Map<String,String> mappings = new HashMap<String, String>();
+		Map<String,String> mappings = new ConcurrentHashMap<>();
 		try
 		{
 			String unzippedFile = this.unzipFile(this.pathToFile);
 			AtomicInteger lineCount = new AtomicInteger(0);
-			try
+
+			try(Stream<String> lineStream = Files.lines(Paths.get(unzippedFile), StandardCharsets.ISO_8859_1))
 			{
-				//I added ".sequential()" because I have line counter and I want to ensure that it works properly.
-				Files.lines(Paths.get(unzippedFile)).filter(p -> !p.startsWith("#")).sequential().forEach( line ->
+				// Use parallel streams at your own peril! I kept running out of RAM and
+				// crashing on my OICR laptop - the server or the office-based workstation can
+				// *probably* handle it but since I can't successfully test, I will leave this
+				// as a sequential stream, for now. But honestly, it took my laptop ~7 minutes
+				// to process the file. I can't imagine a parallel stream would save more
+				// than 3 or 4 minutes...
```

Comment:
Is this newer COSMIC file much bigger? I don't notice the file processing for the current cosmic file taking anywhere near as long as this

Reply:
I can't recall how previous one was, but yes. This newer one is big.